### PR TITLE
Warn users when collisions are not supported

### PR DESCRIPTION
**Pull request by fabriceci**
Some collisions are not supported in the Physics engine, this PR adds a warning to the user.

### CODE REVIEW

**Comment by akien-mga** [comment]
Are these warnings that users would see when intentionally attempting such collisions? Or would they show up when they try to just collide bodies without thinking too much and then get warnings that are difficult to understand, as they don't point to which shapes are problematic?

**Comment by fabriceci** [comment]
The warnings appear when these types of collisions occur (in game).

e.g. , if the user uses two concave or soft bodies and try to make a collision between them, it will not work, the message will be there to give the information.

I have indicated in the messages the types of collisions not supported: "between concave segments, word bodies, etc.".

File: servers/physics_2d/godot_collision_solver_2d.cpp [hunk 1]
```diff
@@ -249,6 +251,7 @@ bool GodotCollisionSolver2D::solve(const GodotShape2D *p_shape_A, const Transfor
 
 	} else if (concave_B) {
 		if (concave_A) {
+			WARN_PRINT_ONCE("Collisions between two concave segments are not supported.");
```
Comment (rburing) [inline servers/physics_2d/godot_collision_solver_2d.cpp:254]: This should say "between two concave shapes" for clarity (and the same in 3D).

**Review by rburing — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**